_posts/2017-10-20-yargs-10.md

```diff
@@ -0,0 +1,133 @@
+* there was a major disconnect between how you described the root-level
+  entry-point to your application and the subcommands that your application
+  accepts. `.usage()` hadn't ever been updated to reflect a world in-which
+  commands exist.
+
+The goal of `yargs@10` was to address the above issues.
+
+## .positional(&lt;name&gt;, &lt;options&gt;)
+
+The new [`.positional()`](https://github.com/yargs/yargs/blob/master/docs/api.md#positionalkey-opt) method allows you to provide the same parsing hints
+for positional arguments that you could already provide for flag arguments.
+Let's look at an updated version of our example application:
+
+<pre>
+  <code class="hljs language-javascript">require('yargs')
+  .command('append &lt;w1&gt; &lt;w2&gt;', 'append two strings', (yargs) => {
+    yargs.positional('w1', {
+      describe: 'the left-hand word',
+      type: 'string'
+    }).positional('w2', {
+      describe: 'the right-hand word',
+      type: 'string'
+    })
+  }, (argv) => {
+    console.log(argv.w1 + argv.w2)
+  })
+  .command('add &lt;n1&gt; &lt;n2&gt;', 'add two numbers', (yargs) => {
+    yargs.positional('n1', {
+      describe: 'the left-hand number',
+      type: 'number'
+    }).positional('n2', {
+      describe: 'the right-hand number',
+      type: 'number'
+    })
+  }, (argv) => {
+    console.log(argv.n1 + argv.n2)
+  })
+  .argv</code></pre>
+
+To provide parsing hints for positional arguments, simply use the method
+`.positional()` in your command's [builder function](https://github.com/yargs/yargs/blob/master/docs/api.md#commandmodule). Positional arguments
+now go through the same parser that parses flag arguments and behavior
+will be identical.
+
+## .usage(&lt;cmd&gt;, [description], [builder], [handler])
+
+If you provide more than one argument to the [`.usage()`](https://github.com/yargs/yargs/blob/master/docs/api.md#usagemessagecommand-desc-builder-handler) method, `.usage()`
+now acts as an alias for a [default command](https://github.com/yargs/yargs/blob/master/docs/advanced.md#default-commands). This allows you to configure a command
+that runs as the main entry-point to your application.
+
+Let's look at an example:
+
+<pre>
+  <code class="hljs language-javascript">require('yargs')
+  .usage('$0 &lt;name&gt;', 'say hello', (yargs) => {
+    yargs.positional('name', {
+      describe: 'the person to say hello to',
+      type: 'string'
+    })
+  }, (argv) => {
+    console.info(`hello ${argv.name}`)
+  })
+  .argv</code></pre>
+
+You can now use the same powerful command DSL for
+describing the root-entry point to your application as you use for subcommands!
+
+## Upgrading
+
+There have been a couple breaking changes in this release:
+
+* `.usage()` previously accepted an object as a second argument, treating this
+  as an alias for `.option([obj])`.
+* defining hidden options now has a [slightly different API](https://github.com/yargs/yargs/pull/962).
```

Comment:
This needs more context, maybe a brief example? The pull request is very poorly titled, I had to read the whole thread before I realized what _actually_ happened.

Reply:
Which, as near as I can tell (since it's not documented anywhere but a couple test cases): Hidden options now require `hidden: true` in their config, _not_ the previously implicit `describe: false`.

Reply:
how about: 

```md
* the API for defining hidden options has been made more explicit; rather than
  a falsy description representing a hidden option, the configuration value
  `hidden: true` should be set.
```

Reply:
Sounds good!